src/coord/dist_aggregate.c

```diff
@@ -165,14 +165,25 @@ static void netCursorCallback(MRIteratorCallbackCtx *ctx, MRReply *rep) {
 RSValue *MRReply_ToValue(MRReply *r) {
   if (!r) return RS_NullVal();
   RSValue *v = NULL;
+  size_t l;
   switch (MRReply_Type(r)) {
-    case MR_REPLY_STATUS:
-    case MR_REPLY_STRING: {
-      size_t l;
+    case MR_REPLY_STATUS: {
       const char *s = MRReply_String(r, &l);
       v = RS_NewCopiedString(s, l);
       break;
     }
+    case MR_REPLY_STRING: {
+      double d = 0;
+      // Try to convert to double because RedisModule_Reply_Double() uses
+      // RedisModule_ReplyWithDouble() which converts double into string
+      if (MRReply_ToDouble(r, &d)) {
+        v = RS_NumVal(d);
```

Comment:
So it related to RESP2 "limitation"
https://github.com/redis/redis/blob/74b289a0e12f9f65a6daeec6a66cadc76792f644/src/module.c#L3481C2-L3482C3
```
 * In RESP3 the string is tagged as a double, while in RESP2 it's just a plain string 
 * that the user will have to parse.
```